lib/iris/tests/unit/concatenate/test_concatenate.py

```diff
@@ -242,6 +242,14 @@ def test_datatype_difference_message(self):
         with self.assertRaisesRegex(ConcatenateError, exc_regexp):
             _ = concatenate([cube_1, cube_2], True)
 
+    def test_dim_coords_overlap_message(self):
+        cube_1 = self.cube
+        cube_2 = cube_1.copy()
+        cube_2.coord("time").points = np.arange(1, 3, dtype=np.float32)
+        exc_regexp = "Dimensional coordinate extents differ"
+        with self.assertRaisesRegex(ConcatenateError, exc_regexp):
+            _ = concatenate([cube_1, cube_2], True)
```

Comment:
It is also worth testing on a case where concatenation fails due to bounds rather than points. See these tests https://github.com/SciTools/iris/blob/2a9e131875be42c39ab4b8fc9956e4a4d59f83a2/lib/iris/tests/test_concatenate.py#L352-L370
(I find it odd that these tests aren't failing while the above two are, I think this might warrent investigating)

Reply:
Looking into this, the points tests fail because they try to raise "warnings.warm" not "warnings.warn" from my typo. The bounds tests fail earlier when you compare dimensional coords 

`*** iris.exceptions.ConcatenateError: failed to concatenate into a single cube.
  Dimension coordinates metadata differ: x != x`

So they never get to the new warning code in register to begin with, because Iris doesn't try to concatenate them to start with. 

Also - should the warnings just be captured in the above tests or new ones written for them.

Reply:
Added pytest tests for warnings in these tests

Reply:
Okay, I've looked into these tests, I think they can be "fixed" by changing the line https://github.com/SciTools/iris/blob/2a9e131875be42c39ab4b8fc9956e4a4d59f83a2/lib/iris/tests/test_concatenate.py#L357 and
https://github.com/SciTools/iris/blob/2a9e131875be42c39ab4b8fc9956e4a4d59f83a2/lib/iris/tests/test_concatenate.py#L367
to have numpy arrays of dtype float32, e.g. having:
```
np.array([[2.5, 0.5], [0.5, -0.5]], dtype=np.float32)
```
My instinct here is to modify the existing test since then it serves the purpose of both testing the warning and also ensures that the correct piece of code is being tested here (which it appears was not the case previously).

Reply:
Sounds good - I'm signing off now, so leave the rest in your hands.